### PR TITLE
Templates variables

### LINKED ISSUE #719: Add set and set-file flags to step certificate commands
### Description

Those commands that have a --template flag in the `step certificate` command group should have a way to pass variables using `--set` and `--set-file` flags. The specific commands are `step certificate create` and `step certificate sign`.

### PR DESCRIPTION
**Pull request by maraino**
### Description

This PR allows to use user variables in the templates used in `step certificate create` and `step certificate sign`.

Fixes #719

### CODE REVIEW

File: command/certificate/create.go [hunk 1]
```diff
@@ -263,21 +264,32 @@ $ step certificate create --template root.tpl \
 '''
 
 Create an intermediate certificate using the previous root. This intermediate
-will be able to sign also new intermediate certificates:
+will be able to sign also new intermediate certificates. We will also make the
```
Comment (dopey) [inline command/certificate/create.go:267]:
```suggestion
This intermediate will be able to sign leaf or intermediate certificates.
```

File: command/certificate/create.go [hunk 2]
```diff
@@ -263,21 +264,32 @@ $ step certificate create --template root.tpl \
 '''
 
 Create an intermediate certificate using the previous root. This intermediate
-will be able to sign also new intermediate certificates:
```
Comment (dopey) [inline command/certificate/create.go:266]: By extending the 'maxPathLen` we are enabling this intermediate sign leaf and intermediate certificates.

**Comment by LecrisUT** [comment]
Oh so `--set` simply creates an object accessed via `.Insecure.User`. Can't we have it that it completely overwrites any of the template data? This part is only used in `step certificate` so presumably the user wants much more control than `step ca`, so it shouldn't be "Insecure"

File: command/certificate/create.go [hunk 3]
```diff
@@ -263,21 +264,32 @@ $ step certificate create --template root.tpl \
...
 		"isCA": true,
 		"maxPathLen": 1
 	}
 }
+$ cat organization.json
+{
+	"country": "US",
+	"organization": "Acme Corporation",
+	"organizationUnit": "HQ"
+}
```
Comment (dopey) [inline command/certificate/create.go:289]: You're setting `organizationUnit` in the file and on the cmd line.

Comment (maraino) [reply command/certificate/create.go:289]: Yes, on purpose, the --set flag overwrites the one in the file

**Comment by maraino** [comment]
> Oh so --set simply creates an object accessed via `.Insecure.User`. Can't we have it that it completely overwrites any of the template data? This part is only used in step certificate so presumably the user wants much more control than step ca, so it shouldn't be "Insecure"

For consistency with other commands we should keep the --set, --set-file, and `.Insecure.User` as they are, so you can use the same templates.

Regarding using a flag to manually edit the configuration, that's a nice option, it won't be `--set` but it can be something similar.

**Comment by maraino** [comment]
@LecrisUT can you add a new enhancement for that?

File: command/certificate/create.go (hunk 2 above)
Comment (maraino) [reply command/certificate/create.go:266]: Fixed with 57b8a06

**Comment by LecrisUT** [comment]
With the hints from this PR, I think I can come up with something of that sort. Just need to edit `data` with the appropriate input. I'll see what I can do this weekend

**Review by dopey — APPROVED** [review]
lgtm

**Comment by maraino** [comment]
> With the hints from this PR, I think I can come up with something of that sort. Just need to edit data with the appropriate input. I'll see what I can do this weekend

@LecrisUT using `data` won't work directly, because that's the information available to the template, you want something that works after rendering the template, either on the template itself (probably easier) or over the resulting x509util.Certificate or x509util.CertificateRequest.

**Pull request merged**